ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java

```diff
@@ -115,36 +169,83 @@ private void showWelcomeMessage() {
   }
 
   private void executeStatements(final String queries) {
-    final List<QueryMetadata> queryMetadataList = ksqlEngine.createQueries(queries, ksqlConfig);
-    for (QueryMetadata queryMetadata : queryMetadataList) {
-      if (queryMetadata instanceof PersistentQueryMetadata) {
-        PersistentQueryMetadata persistentQueryMetadata = (PersistentQueryMetadata) queryMetadata;
-        persistentQueryMetadata.start();
+    final List<Pair<String, Statement>> statementPairs =
+        ksqlEngine.parseStatements(queries, ksqlEngine.getMetaStore().clone());
+    for (final Pair<String, Statement> statementPair: statementPairs) {
+      final String statementString = statementPair.getLeft();
+      final Statement statement = statementPair.getRight();
+      if (functionMap.containsKey(statement.getClass())) {
+        functionMap.get(statement.getClass()).accept(statementPair);
       } else {
         final String message = String.format(
             "Ignoring statements: %s"
-            + "%nOnly CREATE statements can run in standalone mode.",
-            queryMetadata.getStatementString()
+                + "%nOnly DDL (CREATE STREAM/TABLE, DROP STREAM/TABLE, SET, UNSET) "
+                + "and DML(CSAS, CTAS and INSERT INTO) statements can run in standalone mode.",
+            statementPair.getLeft()
         );
         System.err.println(message);
         log.warn(message);
       }
     }
+
+  }
+
+
+  private void handlePersistentQuery(
+      final Statement statement,
+      final String statementString,
+      final Map<String, Object> configProperties) {
+    final Query query;
+    if (statement instanceof CreateAsSelect) {
+      query = ((CreateAsSelect) statement).getQuery();
+    } else if (statement instanceof InsertInto) {
+      query = ((InsertInto) statement).getQuery();
+    } else {
+      throw new KsqlException("Only CSAS/CTAS and INSERT INTO are persistent queries: "
+          + statementString);
+    }
+
+    final QueryMetadata queryMetadata =
+        ksqlEngine.getQueryExecutionPlan(query, ksqlConfig);
+    if (statement instanceof CreateAsSelect) {
```

Comment:
I think this if statement is not necessary (though could be wrong).

Just rename `validateCsasCtas` to `validate` and change it to take a `Statement`...